test/extensions/common/wasm/test_data/test_cpp_null_plugin.cc

```diff
@@ -1,38 +1,15 @@
-#include "extensions/common/wasm/null/null_plugin.h"
+// NOLINT(namespace-envoy)
+#include "include/proxy-wasm/null_plugin.h"
 
-namespace Envoy {
-namespace Extensions {
-namespace Common {
-namespace Wasm {
-namespace Null {
-namespace Plugin {
+namespace proxy_wasm {
+namespace null_plugin {
 namespace CommonWasmTestCpp {
-ThreadSafeSingleton<NullPluginRegistry> null_plugin_registry_;
+NullPluginRegistry* context_registry_;
 } // namespace CommonWasmTestCpp
 
-/**
- * Config registration for a Wasm filter plugin. @see NamedHttpFilterConfigFactory.
- */
-class CommonWasmTestPluginFactory : public NullVmPluginFactory {
-public:
-  CommonWasmTestPluginFactory() {}
+RegisterNullVmPluginFactory register_common_wasm_test_cpp_plugin("CommonWasmTestCpp", []() {
```

Comment:
Just curious, how we register the factory now? I saw you comment the following code. 

```
static Registry::RegisterFactory<CommonWasmTestPluginFactory, NullVmPluginFactory> register_;
```

Reply:
The new registration is by global variable constructor, just like the old one but less complicated because it does not need to be as general as the Envoy version.  In this case, the constructor for RegisterNullVmPluginFactory inserts the provided function into a table which is hung off a pointer (to avoid the initialization sequencing issue).   Since these are global initializers which are single threaded and since table insertions can occur in any order it is safe.